tests/server/api/routers/v1/test_datasets.py

```diff
@@ -322,6 +326,58 @@ async def test_get_dataset_download_specific_version(test_client, dataset_with_r
     assert_frame_equal(actual, expected)
 
 
+async def test_get_dataset_jsonl_openai_ft(test_client, dataset_with_messages: Tuple[int, int]):
+    dataset_id, dataset_version_id = dataset_with_messages
+    dataset_global_id = GlobalID(Dataset.__name__, str(dataset_id))
+    dataset_version_global_id = GlobalID(DatasetVersion.__name__, str(dataset_version_id))
+    response = await test_client.get(
+        f"/v1/datasets/{dataset_global_id}/jsonl/openai_ft?version={dataset_version_global_id}"
+    )
+    assert response.status_code == 200
+    assert response.headers.get("content-type") == "text/plain"
+    assert response.headers.get("content-encoding") == "gzip"
+    assert response.headers.get("content-disposition") == 'attachment; filename="xyz.jsonl"'
+    assert (
+        response.content.decode()
+        == json.dumps(
+            {
+                "messages": [
+                    {"role": "system", "content": "x"},
+                    {"role": "user", "content": "y"},
+                    {"role": "assistant", "content": "z"},
+                ]
+            }
+        )
+        + "\n"
+    )
```

Comment:
Suggestion: If something is off, the pytest diff is much clear if you JSON load the response content and compare dictionaries.

Reply:
updated. thx